src/components/ProfileMatch/index.tsx

```diff
@@ -0,0 +1,159 @@
+'use client';
+
+import { useMemo } from 'react';
+import CONFIG from '@/lib/configs';
+import COLORS from '@/styles/colors';
+import { Flex } from '@/styles/containers';
+import { H3, LinkColored, P } from '@/styles/text';
+import {
+  CaseListing,
+  DocumentTranslation,
+  LimitedCaseAssignment,
+  Listing,
+  Profile,
+} from '@/types/schema';
+import {
+  formatEnumeration,
+  formatTimestamp,
+  timestampStringToDate,
+} from '@/utils/helpers';
+import { useProfile } from '@/utils/ProfileProvider';
+import Icon from '../Icon';
+import { Container, IconDiv } from './styles';
+
+interface MatchField<T extends Listing> {
+  getMatch: (data: T, profileData: Profile) => boolean | undefined;
+  getText: (
+    data: T,
+    profileData: Profile,
+    match: boolean | undefined,
+  ) => string;
+}
+
+const timeCommitmentMatch: MatchField<CaseListing> = {
+  getMatch: (data, profileData) =>
+    data.hours_per_week
+      ? profileData.hours_per_month > data.hours_per_week * 4
+      : undefined,
+  getText: (data, profileData, match) =>
+    data.hours_per_week
+      ? `Your time commitment (${profileData.hours_per_month} hours/month) ${
+          match ? 'meets' : 'does not meet'
+        } the minimum for this case (${data.hours_per_week * 4} hours/month).`
+      : 'No information available about Time Commitment',
+};
+
+const startDateMatch: MatchField<
+  CaseListing | LimitedCaseAssignment | DocumentTranslation
+> = {
+  getMatch: (data, profileData) => {
+    const date =
+      data.listing_type === 'CASE' ? data.upcoming_date : data.deadline;
+    return date
+      ? timestampStringToDate(profileData.start_date) <=
+          timestampStringToDate(date)
+      : undefined;
+  },
+  getText: (data, profileData, match) =>
+    match === undefined
+      ? `No information available about the ${
+          data.listing_type === 'CASE'
+            ? 'Next Court Hearing/Filing Date'
+            : 'Deadline'
+        }`
+      : `The ${
+          data.listing_type === 'CASE'
+            ? 'next court hearing/filing date'
+            : 'deadline'
+        } (${formatTimestamp(
+          data.listing_type === 'CASE' ? data.upcoming_date : data.deadline,
+        )}) is ${
+          match ? 'after' : 'before'
+        } your first available start date (${formatTimestamp(
+          profileData.start_date,
+        )}).`,
```

Comment:
this doesn't give readable vibes...
instead of making this all in "one line", it could be better to split the logic into separate variables and if statements.

Reply:
gotcha, changed.